From 0209a27efb55d5d76b8155b7562ef8b364083335 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Fri, 1 Sep 2017 00:03:49 +0300
Subject: [PATCH 1998/2241] pci-uio: dts: Enable pci-uio driver on A7K PCAC adn
 A8K OCP systems

This driver is relevant only on systems running in PCIe-EP
configuration, therefore it's only enabled on PCAC and OCP boards.

Change-Id: I0370da6b0efeb129e4254f2e6fe495f242c4cb58
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43631
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-pcac.dts | 10 ++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts  | 11 +++++++++++
 2 files changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
index 56248dd..6bdacd2 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
@@ -216,6 +216,16 @@
 			};
 		};
 
+		pci-ep-uio {
+			compatible = "marvell,pci-ep-uio";
+			reg = <0 0 0 0x100000>, <0x80 0x00000000 0x1 0x000000000>;
+			reg-names = "bar0", "host-map";
+			device-id = <0x7080>;
+			class-code = <0x2>;
+			subclass-code = <0x0>;
+			status = "okay";
+		};
+
 		pcie-ep@0x600000 {
 			status = "okay";
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
index c628bd4..2a1f59c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
@@ -135,6 +135,17 @@
 				status = "okay";
 			};
 		};
+
+		pci-ep-uio {
+			compatible = "marvell,pci-ep-uio";
+			reg = <0 0 0 0x100000>, <0x80 0x00000000 0x1 0x000000000>;
+			reg-names = "bar0", "host-map";
+			device-id = <0x7080>;
+			class-code = <0x2>;
+			subclass-code = <0x0>;
+			status = "okay";
+		};
+
 	};
 
 	cpn-110-slave {
-- 
2.7.4

